src/editors/playgroundController.ts

```diff
@@ -140,26 +142,45 @@ export default class PlaygroundController {
         );
 
         if (confirmRunAll !== 'Yes') {
-          return Promise.resolve(false);
+          return resolve(false);
         }
       }
 
       const activeEditor = vscode.window.activeTextEditor;
       const codeToEvaluate = activeEditor?.document.getText() || '';
       let result;
 
-      try {
+      // Show a running progress in the notification area with support for cancellation
+      await vscode.window.withProgress({
+        location: ProgressLocation.Notification,
+        title: 'Running a playground...',
+        cancellable: true
+      }, async (progress, token) => {
+        token.onCancellationRequested(() => {
+          // If a user clicked the cancel button terminate all playground scripts
+          this._languageServerController.cancelAll();
```

Comment:
Couldn't we be executing more than 1 playground at a time? Perhaps something to come back to later.

Reply:
We can open several playgrounds. I create a new Worker for every run of a playground. So it should be a new serviceProvider and mongoClient for each thread. Probably it should be fine 🤔 Or do you think they can somehow interfere each other?